### DISCUSSION TITLE
Question related to 21 October 2020 milestone

### DISCUSSION THREAD

**Discussion by iuriemaxim** [INSPIRE-MIF/helpdesk-validator#786] [General]
As a new deadline is approaching in October, please help me to correctly understand what tests should be passed in order to ensure that metadata, view (WMS) and download services (only WFS) are compliant, to reach the INSPIRE deadline from 21 October 2020.

**Comment by MarcoMinghini** [comment]
Dear @iuriemaxim,
the October 2020 deadline requires that all datasets are conformant to the Implementing Rules on the interoperability of spatial data sets and services. This means that all tests for spatial data sets shall pass (i.e. all data sets shall be harmonized) and that the tests on the metadata associated to data sets shall also satisfy the requirements on interoperability.

I hope this answers your question.

**Comment by iuriemaxim** [comment]
Dear @MarcoMinghini, I would appreciate a more clear answer for a dataset composed of multiple feature types that can be viewed via WMS and downloaded via WFS.

I suppose that there should be three metadata files:
1. metadata of the dataset to be compliant with conformance class 2 for metadata TG version 2.0 or with 2b if ....
2. metadata of the view service of the dataset to be compliant with conformance class 4 for metadata TG version 2.0
3. metadata of the download service of the dataset to be compliant with conformance class 4 for metadata TG version 2.0

There should be two services:
4. The view service should be compliant with Conformance Class WMS or WMTS and layer names should be harmonized.
5. The download service, if WFS, should be compliant with Conformance classes Download Service: Direct WFS and Download Service Predefined WFS.

Trough WFS, in a dataset can be served one or more feature types, but only one dataset should be accessible at a certain endpoint. 
6. For accessing each feature type there is a WFS request, and each should be checked against all 4 conformance classes from Interoperable datasets in GML (Guidelines for the Encoding of Spatial data version 3.3), namely Data consistency, INSPIRE GML application schemas, Information accessibility, Reference systems.

If there are feature types accessible trough WFS corresponding to a data theme for which a specific data theme conformance class exist, then, the WFS request to access that feature type should be tested against the following conformance classes:
7. Application schema, Data consistency, Information accessibility, Reference systems - for the corresponding data theme.
8. GML Application schema - for the corresponding data theme.

XXXXX

If the dataset is not downloadable trough a WFS, but trough ATOM, then the service should be compliant with Download Service - Pre-defined Atom and its content should be compliant with the above points 6 / 7 & 8.

If the dataset is downloadable trough Pre-defined SOS or WCS, then ...

Currently no linkage in between the dataset and services is checked in the ETS, so the data provider should manually verify that the linkage in between metadata files version 2.0 and GetCapabilities requests of the view and download services are ensured. Linkage Checker is ensured for metadata version 1.3, (deprecated) but metadata should be made according to TG version 2.0.

I am expecting something like this above, in order to be clear how to use the validator in order to ensure that the dataset(s) provided conforms with the 21 October 2020 milestone requirement.

**Comment by MarcoMinghini** [comment]
Dear @iuriemaxim,
thank you for your answer.

Apologies for my quick feedback, but from the pure perspective of the Validator - given that almost all the deadlines will expire soon, this obviously means that all the tests should pass. As you know, the Validator provides separate tests for the single resources (metadata, datasets and services), while checking resources together can be done through the linkage checker (in September a new version of the Geoportal and the Linkage Checker will be released will full support for MD TG 2.0).

I will leave this thread open, but I encourage you to move the discussion to the [INSPIRE Community Forum](https://inspire.ec.europa.eu/forum) where you woud probably get more feedback.

**Comment by iuriemaxim** [comment]
@MarcoMinghini Thank you for the information provided. I opened this discution in the validator section in order to understand the role of the INSPIRE ETF validator in the context of the 2020 milestone having in mind that on the linkage checker webpage it is clearly stated that the only official validator is the ETF one and not the linkage checker. Probably with the new version of the linkage checker it will be written something else about official validator(s), if the ETF validator will not check the linkage between metadata, services and spatial data.

Can it be confirmed then, that the linkage checker will be using the ETF validator in order to check the conformance of network services, spatial data within the datasets and metadata of the datasets/services?

Ir can be confirmed if the linkage cheker will check also the linkage of non-network services (Invocable/Interoperable/Harmonised Spatial Data Services) or both the ETF validator and the linkage cheker will check only the network services and not the data served by the Invocable Spatial Data Services even if they are harmonised or just interoperable?

**Comment by iuriemaxim** [comment]
@MarcoMinghini This topic is still not clear, so any further information is appreciated.

**Comment by iuriemaxim** [comment]
@fabiovinci There are any news related to this question as it is still valid even today?

**Comment by fabiovinci** [comment]
Dear @iuriemaxim,

regarding the "_21 October 2020 milestone_", as you know, the validator now includes the tests for all data themes, therefore the compliance of datasets to IR-ISDSS can be easily checked.

As for your other questions:

> Can it be confirmed then, that the linkage checker will be using the ETF validator in order to check the conformance of network services, spatial data within the datasets and metadata of the datasets/services?

At the moment, there is no planning to integrate the linkage checker tool with the INSPIRE Reference Validator, which does not check the linkage between metadata, services and spatial data. This functionality is delegated to the Geoportal which is now in a revision phase, due to the change in its technology (the new Geoportal will be based on GeoNetwork) and to the work done for the "[Good Practice: Data and Service Linking Simplification](https://github.com/INSPIRE-MIF/gp-data-service-linking-simplification)".

> Ir can be confirmed if the linkage checker will check also the linkage of non-network services (Invocable/Interoperable/Harmonised Spatial Data Services) or both the ETF validator and the linkage checker will check only the network services and not the data served by the Invocable Spatial Data Services even if they are harmonised or just interoperable?

Currently there is no planning to integrate any check for the SDS (non-network services -->Invocable/Interoperable/Harmonised Spatial Data Services).

**Comment by iuriemaxim** [comment]
@fabiovinci Thank you so much for the clarifications. As the functionalities for checking the linkages between metadata, services and spatial data are still under development or not planned yet, would be better to keep this issue opened. If possible could be marked with a tag similar to "response necessary when changes occurs" or "open question"?

**Comment by fabiovinci** [comment]
Dear @iuriemaxim,

if there will be the integration of this function in the Reference Validator, it will be communicated through an "_Announcements_" (see [this discussion](https://github.com/INSPIRE-MIF/helpdesk-validator/discussions/763) for more references).
I will convert the issue into a discussion to leave it open for further comment.
